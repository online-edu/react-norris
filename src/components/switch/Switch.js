import React from 'react';
import PropTypes from 'prop-types';

/**
 * Switch component.
 */
const Switch = ({ switchToggle, id, label }) => (
  <div className="custom-control custom-switch">
    <input
      type="checkbox"
      className="custom-control-input"
      aria-label={`${label} Checkbox`}
      id={id}
      name={id}
      onChange={switchToggle}
    />
    <label
      className="custom-control-label"
      htmlFor={id}
      aria-label={`${label} checkboz`}
      aria-describedby={id}
    >
      {label}
    </label>
  </div>
);
/**
 * Switch props types.
 */
Switch.propTypes = {
  /** Id to be assigned to the control. */
  id: PropTypes.string.isRequired,
  /** Display value */
  label: PropTypes.string.isRequired,
  /** Gets called when the user clicks on switch */
  switchToggle: PropTypes.func.isRequired,
};
export default Switch;
