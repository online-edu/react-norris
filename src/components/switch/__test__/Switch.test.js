import React from 'react';
import { shallow, mount } from 'enzyme';
import Switch from '../Switch';

describe('<Switch />', () => {
  const onSwitchClick = jest.fn();
  const wrapper = (
    <Switch id="random" label="Random" switchToggle={onSwitchClick} />
  );

  it('renders correctly', () => {
    const component = shallow(wrapper);
    expect(component).toMatchSnapshot();
  });

  it('contains class name "custom-control, custom-switch" ', () => {
    const component = shallow(wrapper);
    expect(component.find('.custom-control')).toHaveLength(1);
    expect(component.find('.custom-switch')).toHaveLength(1);
  });

  it('allows us to set props', () => {
    const component = mount(wrapper);
    expect(component.props().label).toEqual('Random');
    component.setProps({ label: 'Jokes' });
    expect(component.props().label).toEqual('Jokes');
  });
});
