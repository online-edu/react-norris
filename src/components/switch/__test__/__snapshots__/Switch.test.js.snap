// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> renders correctly 1`] = `ShallowWrapper {}`;
